### DISCUSSION TITLE
Rail noise sources - impact noise and contact filter

### DISCUSSION THREAD

**Discussion by espen-wold-adsk** [Universite-Gustave-Eiffel/NoiseModelling#677] [Q&A] [answered]
Hi,

I am looking at your code - and in particular some of your unit tests - in an attempt to use it as an independent verification for the results of my own implementation of CNOSSOS. In the area of rail noise sources I came across a few discrepancies that I would love to get your thoughts on:

- It looks like your code applies the contact filter to the total roughness _after_ adding the contribution from impact noise, while the CNOSSOS documentation seems to imply that the impact noise should be added to the total-roughness-after-application-of-contact-filter (equations IV-6 and IV-10). I verified this by modifying the order of operations in my own code, and thus producing values matching the expected output in some of your tests.

- If I am reading your code correct, you are not making use of a 'JoinDensity' parameter, and rather directly applying what I had interpreted as `L_R, Impact-single` as per equation IV-11 in the CNOSSOS documentation.

I hope you have time to help me understand these discrepancies!

Regards,
Espen

**Comment by pierromond** [comment]
Thank you very much for your comment ! We will look at it and come back to you :)

**Comment by ochiello** [comment, marked as answer]
Hi Espen,

Thank you for your very pertinent comments. You're absolutely right on both issues.

- the contact filter should not be applied to the roughness specific to impact noise; this is an error that we will quickly correct!
- not taking 'joint density' into account is the method recommended for the production of noise maps for the French national rail network (SNCF réseau): the "french" equivalent roughness (L_R_ Impact) was determined with this assumption. 

Just one question: which document are you referring to when you mention the CNOSSOS documentation ?

Regards,
Olivier

**Comment by pierromond** [comment]
Done on 8 october and in NoiseModelling 5.0 ! Thank you
